oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -516,6 +561,23 @@ private APIGatewayProxyResponseEvent generateAuthRedirect(
                             client, configurationService.getInternalSectorUri());
             var state = new State();
             orchestrationAuthorizationService.storeState(session.getSessionId(), state);
+            String reauthenticateClaim;
+
+            try {
+                reauthenticateClaim =
+                        reauthRequested ? getReauthenticateClaim(authenticationRequest) : null;
+            } catch (RuntimeException e) {
+                return generateErrorResponse(
+                        authenticationRequest.getRedirectionURI(),
+                        authenticationRequest.getState(),
+                        authenticationRequest.getResponseMode(),
+                        new ErrorObject(OAuth2Error.INVALID_REQUEST_CODE, e.getMessage()),
+                        ipAddress,
+                        persistentSessionId,
+                        authenticationRequest.getClientID().getValue(),
+                        clientSessionId);
+            }
```

Comment:
Is it possible to pull this out as its repeating the same logic above ?

Reply:
Unfortunately not, because it would have to return two different types - either an error response, or a claim. I tried it but couldn't get it to work